cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/twill/StatefulTwillPreparer.java

```diff
@@ -35,4 +35,12 @@
    */
   StatefulTwillPreparer withStatefulRunnable(String runnableName, boolean orderedStart,
                                              StatefulDisk... statefulDisk);
+
+  /**
+   * Mount the provided disk name for the given runnable name as readonly volume.
+   * @param runnableName name of the {@link TwillRunnable}
+   * @param diskName name of the {@link StatefulDisk}
+   * @return @return this {@link TwillPreparer}
+   */
+  SecureTwillPreparer withReadonlyDisk(String runnableName, String diskName);
```

Comment:
Is this correct to return `SecureTwillPreparer` instead of `StatefulTwillPreparer`?

Reply:
It was a mistake. thanks for catching it.